### PR TITLE
Tganley/feature/ns payloads

### PR DESCRIPTION
**Pull request by tganley**
Tested on hardware in the flatsat. Only known issue is that the xmodem code will require more work and testing.

### CODE REVIEW

**Review by rcunrau — DISMISSED** [review]
Great work! :-)

File: ex2_hal/northern_spirit/equipment_handler/include/northern_voices_games.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * @file northern_voices_games.h
+ * @author Thomas Ganley
+ * @date 2022-06-12
+ */
+
+#ifndef EX2_HAL_NORTHERN_SPIRIT_EQUIPMENT_HANDLER_INCLUDE_NORTHERN_VOICES_GAMES_H_
+#define EX2_HAL_NORTHERN_SPIRIT_EQUIPMENT_HANDLER_INCLUDE_NORTHERN_VOICES_GAMES_H_
+
+#include "northern_spirit_handler.h"
+#include <redposix.h>
```
Comment (Taylorr82) [inline ex2_hal/northern_spirit/equipment_handler/include/northern_voices_games.h:25]: Headers should only have the bare minimum includes, I don't see any reliance edge stuff in a few of these, headers, should remove them

File: ex2_hal/northern_spirit/equipment_handler/include/northern_spirit_io.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * @file northern_spirit_io.c
+ * @author Thomas Ganley
+ * @date 2021-10-26
+ */
+#ifndef NORTHERN_SPIRIT_IO_H
+#define NORTHERN_SPIRIT_IO_H
+
+#include "northern_spirit_handler.h"
+#include "FreeRTOS.h"
```
Comment (Taylorr82) [inline ex2_hal/northern_spirit/equipment_handler/include/northern_spirit_io.h:24]: Lots of unneeded includes here too. Includes not needed by the header but needed in the C file should be only in the C file

**Review by Taylorr82 — CHANGES_REQUESTED** [review]

File: source/base_64.c [hunk 3]
```diff
@@ -0,0 +1,95 @@
+/*
```
[the remark this answers is not in the document]
Comment (tganley) [reply source/base_64.c:1]: Good call. I'll put the xmodem stuff in there too maybe

**Review by Taylorr82 — APPROVED** [review]

**Review by Taylorr82 — APPROVED** [review]

**Pull request merged**